packages/web/src/pages/upload-page/utils/uploadPreviewContext.tsx

```diff
@@ -0,0 +1,64 @@
+import { createContext, memo, useCallback, useState } from 'react'
+
+import { Nullable, playerSelectors, queueActions } from '@audius/common'
+import { useSelector, useDispatch } from 'react-redux'
+
+const { getPlaying } = playerSelectors
+const { pause: pauseQueue } = queueActions
+
+type PreviewContextProps = {
+  playPreview: (preview: HTMLAudioElement, idx: number) => void
+  stopPreview: () => void
+  playingPreviewIndex: number
+}
+
+export const UploadPreviewContext = createContext<PreviewContextProps>({
+  playPreview: () => {},
+  stopPreview: () => {},
+  playingPreviewIndex: -1
+})
+
+export const UploadPreviewContextProvider = memo(
```

Comment:
Sup with memo?

Reply:
i saw that the other context providers had it and figured it was important

Reply:
seems like it it preventing all of the consumers of the context from rerendering too often

Reply:
I don't think so actually, but mind confiming in dev tool?

Reply:
yea i can try to verify, should the memo be removed from all of the other context providers too?